### DISCUSSION TITLE
Importing one slightly diverged library on top of another

### DISCUSSION THREAD

**Discussion by JamesSwift** [photoprism/photoprism#1056] [Q&A] [answered]
Hello. I was wondering if you can help me out with a "what will happen if" question.

For the last 15 years or so I have had a folder on my pc where I've kept my photos. Over the years I have uploaded portions of it to google picasa , then plus, then photos. Some phots have been lost over the years and not been added to both libraries, some photos are in better quality locally.  I intend to switch entirely to photoprism, by importing both libraries and letting it sort them out into one combined folder (I'll keep backups till I'm satisfied it's all there).

I've started by doing a google takeout, running a script to delete all the "-edited" files (assuming there is a matching original file), then importing it into photoprism. It's quite a large library, so I don't want to waste days of importing time by making a mistake in approach.

So, finally, the question. Now that my google takeout has been imported, what will happen if I import my local library? Filenames should be a match (in theory) on any duplicates, but the quality/resolution of some duplicates will differ.

Will photoprism store a duplicate of every file? Will it discard the duplicates? If so and if they differ in resolution, which resolution will it keep? I don't mind doing a little cleanup of course, but if I'm likely going to end up with a big mess, I'd rather write my own script to merge them beforehand.

Thanks for any help you can offer.

**Comment by alexisLefebvre** [comment]
Just as a side note, you can use [czkawka](https://github.com/qarmin/czkawka) to identify and remove duplicated images before importing them.

**Reply by JamesSwift** [reply]
Thanks for the tip, I'll keep that in mind. I'd still like to know how PP will react though, just good to understand it's logic for future reference.

**Reply by alexisLefebvre** [reply]
czkawka can scan 10k photos in some minutes, and allow to delete a file in 2 clicks. It will probably be faster than using the PhotoPrism UI to delete every duplicate.

**Comment by graciousgrey** [comment, marked as answer]
During the import only exact duplicates (files with the same sha1 hash) will be skipped.

If your files do have different sha1 hashes due to different resolutions, they will be imported even if they have the same filename as already existing files. 
So you end up with all files in your library. 

Some of those files with differing resolutions might be stacked if they have the same place and time metadata or the same unique image id. For those you could go through the stacks and delete the version you don't want to keep.

Others e.g. those without geo information will not be stacked.

**Reply by JamesSwift** [reply]
Thank you for the help. I have started the import of the second library now, but I took the precaution of taking a zfs snapshot of the database, storage and originals folders first. So far it looks like your description is accurate, probably 2/3 of my library should have geo data and so hopefully will stack if there are different resolutions. Is there a way to manually stack the other images?

**Reply by graciousgrey** [reply]
No not yet, but it is planned: https://github.com/photoprism/photoprism/issues/28
